compiler/src/lnntoamm/Types.ts

```diff
@@ -1,32 +1,41 @@
-import { LPNode } from "../lp";
-import Scope from "./Scope";
-import { genName, TODO } from "./util";
+import { LPNode } from '../lp';
+import Scope from './Scope';
+import { Equalable, genName, TODO } from './util';
 
 type Fields = {[name: string]: Type | null};
-export type GenericArgs = {[name: string]: Type | null};
+type GenericArgs = {[name: string]: Type | null};
 type TypeName = [string, TypeName[]];
 
 const parseFulltypename = (node: LPNode): TypeName => {
   const name = node.get('typename').t.trim();
   let genericTys: TypeName[] = [];
-  if (node.get('opttypegenerics').has()) {
-    const generics = node.get('opttypegenerics').get();
+  if (node.has('opttypegenerics')) {
+    const generics = node.get('opttypegenerics');
     genericTys.push(parseFulltypename(generics.get('fulltypename')));
-    generics.get('cdr').getAll().map(n => n.get('fulltypename'));
+    genericTys.push(...generics.get('cdr').getAll().map(n => n.get('fulltypename')).map(parseFulltypename));
   }
   return [name, genericTys];
 };
 
-export default abstract class Type {
+// TODO: figure out type aliases (i think it actually makes sense to make a new type?)
```

Comment:
You can't alias a base type, then? Because then `type int = int64` will make a new type and that won't match `add(int64, int64): Result<int64>` (for instance).

Reply:
Well, if there was a `class Alias extends Type` then `compatibleWithConstraint` would allow it to be compatible with the aliased type